README.tfdoc.hcl

```diff
@@ -0,0 +1,545 @@
+        END
+      }
+
+      variable "target_type" {
+        type        = string
+        description = <<-END
+          Type of target that you must specify when registering targets with
+          this target group. The possible values are `instance` (targets are
+          specified by instance ID) or `ip` (targets are specified by IP
+          address) or `lambda` (targets are specified by lambda arn). Note that
+          you can't specify targets for a target group using both instance IDs
+          and IP addresses. If the target type is `ip`, specify IP addresses
+          from the subnets of the virtual private cloud (VPC) for the target
+          group. You can't specify publicly routable IP addresses.
+        END
+      }
+
+      variable "port" {
+        type        = number
+        description = <<-END
+          Port on which targets receive traffic, unless overridden when
+          registering a specific target. Required when `target_type` is
+          `instance` or `ip`. Does not apply when `target_type` is `lambda`.
+        END
+      }
+
+      variable "protocol" {
+        type        = string
+        description = <<-END
+          Protocol to use for routing traffic to the targets. Should be one of
+          `GENEVE`, `HTTP`, `HTTPS`, `TCP`, `TCP_UDP`, `TLS`, or `UDP`.
+          Required when `target_type` is instance, `ip` or `alb`. Does not apply
+          when `target_type` is `lambda`.
+        END
+      }
+
+      variable "protocol_version" {
+        type        = string
+        default     = "HTTP1"
+        description = <<-END
+          Only applicable when protocol is `HTTP` or `HTTPS`. The protocol
+          version. Specify `GRPC` to send requests to targets using gRPC.
+          Specify `HTTP2` to send requests to targets using `HTTP/2`. The
+          default is HTTP1, which sends requests to targets using HTTP/1.1
+        END
+      }
+
+      variable "preserve_client_ip" {
+        type        = bool
+        description = <<-END
+          Whether client IP preservation is enabled.
+        END
+      }
+
+      variable "lambda_multi_value_headers_enabled" {
+        type        = bool
+        description = <<-END
+          Whether the request and response headers exchanged between the load
+          balancer and the Lambda function include arrays of values or strings.
+          Only applies when `target_type` is `lambda`.
+        END
+      }
+
+      variable "load_balancing_algorithm_type" {
+        type        = bool
+        description = <<-END
+          Determines how the load balancer selects targets when routing
+          requests. Only applicable for Application Load Balancer Target Groups.
+          The value is `round_robin` or `least_outstanding_requests`.
+        END
+      }
+
+      variable "health_check" {
+        type        = object(health_check)
+        description = <<-END
+          Determines how the load balancer selects targets when routing
```

Comment:
This isn't the correct description for the `health_check` variable

Reply:
You're right, my bad.